src/components/file/file.tsx

```diff
@@ -67,6 +68,12 @@ export class File {
     @Prop({ reflect: true })
     public label: string;
 
+    /**
+     * Optional file icon.
+     */
+    @Prop()
+    public fileIcon: string;
```

Comment:
Do you think we need the `file` prefix on this property?  What does it add compared to just `icon`? :slightly_smiling_face:

Reply:
Currently nothing, but since we have the option of leading and trailing icons in `limel-input-field`, it doesn't seem impossible that we might add some option for that here as well at some point, and if that happens, it will be helpful that this is already named appropriately. So I think we should keep it. Thoughts?

Reply:
Maybe it shoud be part of the `FileInfo` interface now that I think a bit more about it :thinking:

Reply:
That way we could support multiple files with different icons in the future

Reply:
It is already part of the FileInfo api… and of course, that means we can have the consumer change it when `value` is updated… didn't think of that. Thanks! 😄

Reply:
Ah… no… it's not part of the `FileInfo` interface, it's part of the `Chip` interface, but that's not exposed. I was confusing them.

Yeah, we should add that.

Reply:
@joelricci How urgent is the custom-icon feature for you? I'm just thinking of how I should prioritise this versus some other things in the feature-cycle…

Reply:
@adrianschmidt Probably not super-urgent. Let me check with the team...

Reply:
So it's not very urgent, but would appreciate if someone could keep me updated on the progress as we have other issues and PR we want to do depending on it. Also let me know if I can help or you need any input. Thanks for the help gents!

Reply:
Yeah, absolutely! I would like to get it done asap as well.

Reply:
:zap:

@jgroth Can you please do a final review of this, since I've made quite a few changes to it myself?

Reply:
It looks good! :slightly_smiling_face: